src/core/Akka/Actor/ActorRef.cs

```diff
@@ -131,7 +131,15 @@ protected override void TellInternal(object message, IActorRef sender)
             {
                 if (Interlocked.Exchange(ref status, COMPLETED) == INITIATED)
                 {
-                    _result.TrySetResult(message);
+                    if (message is T t)
```

Comment:
In the past we casted from `object` to `T` - now we explicitly check for the presence of `T` and set the result using it.

This _should_ work with types that are assignable to `T` as well.